src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs

```diff
@@ -366,8 +390,15 @@ private void AnalyzeOperationBlockEnd(OperationBlockAnalysisContext context)
                         out ImmutableDictionary<string, string> properties)
                     {
                         properties = null;
+
+                        // Bail out in following cases:
+                        //   1. End user has configured the diagnostic to be suppressed.
+                        //   2. Symbol has error type, hence the diagnostic could be noised
+                        //   3. Symbol is static. For example, assignment to static locals in VB
+                        //      is not unnecessary as the assigned value can be used on the next invocation.
                         if (_options.UnusedValueAssignmentSeverity == ReportDiagnostic.Suppress ||
-                            symbol.GetSymbolType().IsErrorType())
+                            symbol.GetSymbolType().IsErrorType() ||
+                            symbol.IsStatic)
```

Comment:
i'd prefer to actually check for teh explicit case of a 'static local'.

Reply:
Sure, will do.